tests/functional/object/mcg/test_bucket_policy.py

```diff
@@ -850,6 +874,202 @@ def test_bucket_policy_verify_invalid_scenarios(
+
+        # Get bucket policy on the bucket
+        logger.info(f"Getting Bucket policy on bucket: {obc_obj.bucket_name}")
+        get_policy = get_bucket_policy(mcg_obj, obc_obj.bucket_name)
+        logger.info(f"Got bucket policy: {get_policy['Policy']}")
+
+        # Verify DeleteBucket and putObject operation
+        # in both scenarios: Effect=Allow/Deny
+        if effect == "Allow":
+            # Put Object is allowed
+            logger.info("Writing index data to the bucket")
+            assert s3_put_object(
+                s3_obj=obc_obj,
+                bucketname=obc_obj.bucket_name,
+                object_key="index.html",
+                data=index,
+                content_type="text/html",
+            ), "Failed to put object."
+
+            # Delete bucket get access denied.
+            logger.info(f"Deleting bucket {obc_obj.bucket_name}")
+            try:
+                s3_delete_bucket_website(s3_obj=obc_obj, bucketname=obc_obj.bucket_name)
+                raise UnexpectedBehaviour(
+                    "Failed: Bucket got deleted, expect to get AccessDenied."
+                )
+            except boto3exception.ClientError as e:
+                logger.info(e.response)
+                response = HttpResponseParser(e.response)
+                if response.error["Code"] == "AccessDenied":
+                    logger.info(f"Bucket deleting got {response.error['Code']}")
+                else:
+                    raise UnexpectedBehaviour(
+                        f"{e.response} received invalid error code "
+                        f"{response.error['Code']}"
+                    )
+        if effect == "Deny":
+            # Put Object get access denied.
+            logger.info("Writing index data to the bucket")
+            try:
+                s3_put_object(
+                    s3_obj=obc_obj,
+                    bucketname=obc_obj.bucket_name,
+                    object_key="index.html",
+                    data=index,
+                    content_type="text/html",
+                )
+                raise UnexpectedBehaviour(
+                    "Failed: Completed put object to bucket, expect to get AccessDenied."
+                )
+            except boto3exception.ClientError as e:
+                logger.info(e.response)
+                response = HttpResponseParser(e.response)
+                if response.error["Code"] == "AccessDenied":
+                    logger.info(f"PutObject got {response.error['Code']}")
+                else:
+                    raise UnexpectedBehaviour(
+                        f"{e.response} received invalid error code "
+                        f"{response.error['Code']}"
+                    )
+
+            # Delete bucket is allowed.
+            logger.info(f"Deleting bucket {obc_obj.bucket_name}")
+            assert s3_delete_bucket_website(
+                s3_obj=obc_obj, bucketname=obc_obj.bucket_name
+            ), "Failed to delete bucket."
```

Comment:
We should be able to repeat ourselves less here with the following approach:

```suggestion
logger.info("Writing index data to the bucket")
put_op_denied = False
try:
    s3_put_object(
        s3_obj=obc_obj,
        bucketname=obc_obj.bucket_name,
        object_key="index.html",
        data=index,
        content_type="text/html",
    )
except boto3exception.ClientError as e:
    logger.info(e.response)
    response = HttpResponseParser(e.response)
    if response.error["Code"] == "AccessDenied":
        logger.info(f"PutObject got {response.error['Code']}")
        put_op_denied = True
    else:
        raise UnexpectedBehaviour(
            f"The PutObj operation has failed with an unexpected error:"
            f"{e.response} received invalid error code "
            f"{response.error['Code']}"
        )
assert (effect == "Deny" and put_op_denied) or (effect == "Allow" and not put_op_denied),
f"Access to the PutObj operation failed to meet the expectation from applying policy {bucket_policy}"

delete_bucket_op_denied = False
logger.info(f"Deleting bucket {obc_obj.bucket_name}")
try:
    s3_delete_bucket_website(s3_obj=obc_obj, bucketname=obc_obj.bucket_name)
except boto3exception.ClientError as e:
    logger.info(e.response)
    response = HttpResponseParser(e.response)
    if response.error["Code"] == "AccessDenied":
        logger.info(f"Bucket deleting got {response.error['Code']}")
        delete_bucket_op_denied = True
    else:
        raise UnexpectedBehaviour(
            f"The DeleteBucket operation has failed with an unexpected error:"
            f"{e.response} received invalid error code "
            f"{response.error['Code']}"

assert (effect == "Deny" and not delete_bucket_op_denied) or (effect == "Allow" and delete_bucket_op_denied),
f"Access to the DeleteBucket operation failed to meet the expectation from applying policy {bucket_policy}"
```

Reply:
Note that this is a pretty semantic suggestion, so it's up to you whether you want to use it or not

Reply:
Nice suggesstion :-) But I'm using the current one for now.